### ISSUE TITLE
Question/Discussion: Open Assets + Payment Hubs

### ISSUE THREAD

**Issue by Mike-EEE** [MetacoSA/NBitcoin#470]
Hello Community,

I have had my head in this space for a few weeks now, and feel *slightly* educated on this topic, but maybe still swimming a bit, so please bear with me. 😆

My question has to do with this article I just read:
http://allcoinsnews.com/2016/03/26/2016-bitcoin-improvements-enables-open-assets-protocol-scalability/

I am primarily interested in how this space has shaken out since the article in 2016.  Can Payment Hubs be used anywhere?  I did a search and nothing obvious returned.  From what I understand, the Lightning Network is also currently under development and has not been deployed yet, so this is not really a consideration either.

It would seem that the [Open Asset Protocol is used in NBitcoin to issue assets](https://programmingblockchain.gitbooks.io/programmingblockchain/content/other_types_of_asset/issuing_an_asset.html).  I basically wanted to confirm this and see if this is the most up to date and preferred way of doing so.

Further, I am also interested in leveraging this functionality in multiple coins.  That is, I am interested in keeping track of assets on Litecoin and Bitcoin Cash as well as Ethereum (FWIW all of these are featured in Coinbase's commerce model).  I know that there is an [alt coin version](https://github.com/MetacoSA/NBitcoin/tree/master/NBitcoin.Altcoins) but wanted to verify that this colored coins/OAP functionality works for these desired coins.

Finally, I am also interested in having this work on Ethereum, but from what I can tell that is a whole other ball game and would involve a custom/smart contract to make this work.

FWIW, it would seem that BlockCypher has support for an Asset API, [along with support for BTC, LTC, and ETH](https://www.blockcypher.com/dev/bitcoin/#introduction).  Further inspection, however, shows that their Asset API is only in their BTC documentation (I have an email to them to verify).

I would appreciate any and all insight and clarification around this space that can be provided.  I also wanted to share my appreciation and respect for putting some solid .NET into the BTC space with this project as well. :)

Thank you for your efforts here,
Michael

**Comment by NicolasDorier** [comment]
I think for everything asset related you should wait. There has been big advances thanks to Peter Todd in this subject, and giacomo zucco is trying to redact a protocol out of it. (https://www.reddit.com/r/Bitcoin/comments/8wrtmo/assets_on_bitcoin_rbg_protocol_giacomo_zucco/)

This is quite game changing, so I advise you to hold off any attempt to use tokens on Bitcoin for now.

**Comment by Mike-EEE** [comment]
Hey, from the man himself. :)  Thank you sir for your great work here, @NicolasDorier.  It is much appreciated.  As well is the link that you provided.

![](https://media.giphy.com/media/EldfH1VJdbrwY/giphy.gif)

Very insightful stuff.  Seems as if I stumbled upon this topic/space at the right time.  I will check out https://github.com/rgb-org/spec and see what I can find there.

Also, I will also update this thread with any interesting news/notes that I derive from that.  Seems like the .NET space is wide open everywhere these days. 😆 Thanks again!  Closing for now.
